src/arcade/patch/agent/cell/PatchCell.java

```diff
@@ -477,75 +474,60 @@ public PatchLocation selectBestLocation(
      * Find free locations in the neighborhood of the cell.
      *
      * @param sim the simulation instance
-     * @param add true if an additional cell is being added
+     * @param proliferationFlag true if an additional cell is being added
      * @return a list of free locations
      */
-    public Bag findFreeLocations(Simulation sim, boolean add) {
+    public Bag findFreeLocations(Simulation sim, boolean proliferationFlag) {
         Bag freeLocations = new Bag();
-
         PatchLocation currentLocation = this.location;
-        double targetVolume;
-        if (add) {
-            targetVolume = this.getVolume() * 0.5;
-        } else {
-            targetVolume = this.getVolume();
-        }
+        double targetVolume = proliferationFlag ? volume * 0.5 : volume;
```

Comment:
Can we just use the cell `flag` directly, e.g. something like:

```java
boolean proliferationFlag = this.flag == Flag.PROLIFERATIVE;
```

Or is there a good reason to pass it separately? The only case I can think of is that the exit from quiescence call is "pretending" to be a proliferation call but would have an `UNDEFINED` flag which could be addressed by (1) setting the flag or (2) making the check `this.flag != Flag.MIGRATORY` instead.

Reply:
I thought about this approach, but it felt slightly harder to maintain if we were to add another type of cell that maybe randomly through the environment or something like that, I was also thinking it would have to be more explicit about which state. I may be thinking more from a functional programming perspective though.